lib/PAUSE/dist.pm

```diff
@@ -1063,6 +1068,32 @@ sub extract_readme_and_meta {
   }
 }
 
+sub write_updated_meta6_json {
+  my($self, $metafile, $MLROOT, $dist, $sans) = @_;
+
+  my $json = JSON::XS->new->utf8->canonical->pretty;
+
+  open my $meta_fh, '<', $metafile
+    or $self->verbose(1,"Failed to open META6.json file for reading $!");
+  my $meta = eval {
+    $json->decode(join '', <$meta_fh>);
+  };
+  if ($@) {
+    $self->verbose(1,"Failed to parse META6.json file: $@");
+    File::Copy::copy $metafile, "$MLROOT/$sans.meta";
+    return;
+  }
+  close $meta_fh;
+
+  $meta->{'source-url'} = $PAUSE::Config->{PUB_MODULE_URL} . "$MLROOT/$dist";
```

Comment:
Uploading a Perl6 distribution ended up with source-url set as: http://www.cpan.org/authors/id//home/ftp/pub/PAUSE/authors/id/U/UG/UGEXE/Perl6/Text-Levenshtein-Damerau-0.1.7.tar.gz

Maybe the prefixing part of `$MLROOT` needs to be replaced with `PUB_MODULE_URL` instead of concatenating them?